modules/javafx.controls/src/main/java/javafx/scene/control/Skin.java

```diff
@@ -28,9 +28,20 @@
 import javafx.scene.Node;
 
 /**
- * Base class for defining the visual representation of user interface controls
- * by defining a scene graph of nodes to represent the skin.
- * A user interface control is abstracted behind the {@link Skinnable} interface.
+ * An interface for defining the visual representation of user interface controls.
+ * <p>
+ * A Skin implementation should generally avoid modifying its control outside of
+ * {@link #install()} method.  The life cycle of a Skin implementation
+ * is as follows:
+ * <ul>
+ * <li>instantiation
+ * <li>configuration, such as passing of dependencies and parameters
+ * <li>inside of {@link Control#setSkin(Skin)}:
+ * <ul>
+ * <li>uninstalling of the old skin via its {@link #dispose()} method
+ * <li>installing of the new skin via {@link #install()}
+ * </ul>
```

Comment:
- this interface is unrelated to any specific implementation of a Skinnable, at least replace control.setSkin with skinnable.setSkin
- technically, it can't be the setSkin method to have any workload (as already discussed and fixed elsewhere)

aside from these nit-pickings .. don't we overspecify here (and in the doc of install)? From scratch implementations of a Skinnable/Skin pair might decide to do the complete wiring outside of the property, something like

     Skinnable skinnable = new .. 
     Skin skin = new ...
     if (skinnable.getSkin() != null) skinnable.getSkin().dispose();
     skin.install();
     skinnable.setSkin(skin);

Reply:
I don't think we would want to go out of our way to enable this, so I prefer the tighter definition of the life-cycle that Andy is proposing. It seems better to have the Control always call `dispose` and `install` rather than provide an option where the application would call it.

Reply:
I agree with @kevinrushforth , it's one of the cases when the method is public, but it should not be called by an application, only by its Control.

Perhaps we should further clarify this fact?

Reply:
Control is _our_ specific implementation - others might differ :) And not talking about application code but a possibly complete skinnable/skin controlling scheme. 

But I see your point - how can we clearly express that the complete dispose-install cycle should be in the hands of exactly one collaborator (whether a skinnable's property or somewhere else)?

The concrete cells have public methods that are marked as "expert api" - maybe we could do something similar here?

Reply:
I'm not sure that we need to provide for other implementations of Skinnable/Skin. If we can figure out a good way to express the needed invariants in a more neutral way, that's OK, but I'd rather err on the side of clearly defining the requirements of our implementation of Skinnable/Skin.